.github/workflows/repackage-native-libs.yml

```diff
@@ -0,0 +1,183 @@
+        id: cli_ver
+        run: |
+          #echo ${GITHUB_REF#refs/tag/}
+          tree .
+          mv ./*/*.tar.gz .
+          ver=$(ls ./*.tar.gz | xargs basename | sed -E 's/signal-cli-(.*).tar.gz/\1/')
+          echo $ver
+          echo "::set-output name=signal_cli_version::${ver}"
+          tar -xzf ./*.tar.gz
+
+      - name: Get signal-client jar version
+        id: lib_ver
+        run: |
+          JAR_PREFIX=signal-client-java-
+          jar_file=$(find ./signal-cli-*/lib/ -name "$JAR_PREFIX*.jar")
+          jar_version=$(echo "$jar_file" | xargs basename | sed "s/$JAR_PREFIX//; s/.jar//")
+          echo "$jar_version"
+          echo "::set-output name=signal_client_version::$jar_version"
+
+      - name: Download signal-client builds
+        env:
+          RELEASES_URL: https://github.com/signalapp/libsignal-client/releases/download/
+          FILE_NAMES: signal_jni.dll libsignal_jni.dylib
+          SIGNAL_CLIENT_VER: ${{ steps.lib_ver.outputs.signal_client_version }}
+        run: |
+          for file_name in $FILE_NAMES; do
+            curl -sOL "${RELEASES_URL}/v${SIGNAL_CLIENT_VER}/${file_name}"  # note: added v
+          done
+          tree .
+
+      - name: Replace Windows lib
+        env:
+          SIGNAL_CLI_VER: ${{ steps.cli_ver.outputs.signal_cli_version }}
+          SIGNAL_CLIENT_VER: ${{ steps.lib_ver.outputs.signal_client_version }}
+        run: |
+          mv signal_jni.dll libsignal_jni.so
+          zip -u ./signal-cli-${SIGNAL_CLI_VER}/lib/signal-client-java-${SIGNAL_CLIENT_VER}.jar  ./libsignal_jni.so
+          tar -czf signal-cli-${SIGNAL_CLI_VER}-Windows.tar.gz signal-cli-${SIGNAL_CLI_VER}/
+
+      - name: Replace MacOS lib
+        env:
+          SIGNAL_CLI_VER: ${{ steps.cli_ver.outputs.signal_cli_version }}
+          SIGNAL_CLIENT_VER: ${{ steps.lib_ver.outputs.signal_client_version }}
+        run: |
+          jar_file=./signal-cli-${SIGNAL_CLI_VER}/lib/signal-client-java-${SIGNAL_CLIENT_VER}.jar
+          zip -d "$jar_file" libsignal_jni.so
+          zip "$jar_file" libsignal_jni.dylib
+          tar -czf signal-cli-${SIGNAL_CLI_VER}-MacOS.tar.gz signal-cli-${SIGNAL_CLI_VER}/
+
+      - name: Create release
+        id: create_release
+        uses: actions/create-release@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          tag_name: v${{ steps.cli_ver.outputs.signal_cli_version }}  # note: added `v`
+          release_name: v${{ steps.cli_ver.outputs.signal_cli_version }}  # note: added `v`
+          draft: true
+
+      - name: Upload Linux archive
+        uses: actions/upload-release-asset@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          upload_url: ${{ steps.create_release.outputs.upload_url }}
+          asset_path: signal-cli-${{ steps.cli_ver.outputs.signal_cli_version }}.tar.gz
+          asset_name: signal-cli-${{ steps.cli_ver.outputs.signal_cli_version }}-Linux.tar.gz
```

Comment:
I think I'd prefer the names to be lower case, then you don't need to capitalize them below.

Reply:
Ok, then I'd suggest to just go with the $RUNNER_OS names.

Reply:
Sounds good!